pkg/operator/api/api_test.go

```diff
@@ -260,6 +251,50 @@ func TestComponentUpdate(t *testing.T) {
 
 		assert.Equal(t, 1, mockSidecar.Calls)
 	})
+
+	t.Run("sidecar is updated multiple times without data race", func(t *testing.T) {
+		c := componentsapi.Component{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "ns1",
+			},
+			Spec: componentsapi.ComponentSpec{},
+		}
+
+		s := runtime.NewScheme()
+		err := scheme.AddToScheme(s)
+		assert.NoError(t, err)
+
+		err = corev1.AddToScheme(s)
+		assert.NoError(t, err)
+
+		client := fake.NewClientBuilder().
+			WithScheme(s).Build()
+
+		mockSidecar := &mockComponentUpdateServer{}
+		api := NewAPIServer(client).(*apiServer)
+		updateIterations := 100
+		go func() {
+			// Send a component update, give sidecar time to register
+			time.Sleep(time.Millisecond * 500)
+
+			for _, connUpdateChan := range api.allConnUpdateChan {
+				for i := 0; i < updateIterations; i++ {
+					connUpdateChan <- &c
```

Comment:
Seems that it updates sidecar many times with the same data in a loop, which is not a race case I think.

Reply:
The data race occurs in `updateComponentFunc` function which receives component from connUpdateChan and sends it to sidecar. So if we send component to connUpdateChan many times in a loop, `updateComponentFunc` will run in different goroutines and then lead to race condition. It does not matter if the component data sent to connUpdateChan is same or not, what do you think?